core/common/src/main/java/alluxio/resource/ResourcePool.java

```diff
@@ -146,9 +145,12 @@ public T acquire(Integer time, TimeUnit unit) {
    * @param resource the resource to be released, it should not be reused after calling this method
    */
   public void release(T resource) {
-    mResources.add(resource);
-    try (LockResource r = new LockResource(mTakeLock)) {
-      mNotEmpty.signal();
+    if (resource != null) {
```

Comment:
I'd be on board, using those annotations would probably uncover some potential null pointer exceptions through FindBugs. We should still javadoc the behavior of null parameters.